### ISSUE TITLE
show_opt_in checkboxes remaining unchecked when calling editor.setValue()

### ISSUE THREAD

**Issue by bilowik** [json-editor/json-editor#807]
### General information

- json-editor version: 2.3.0

### Expected behavior
Setting show_opt_in in editor options gives each field in the editor a checkbox to enable/disable that field in the json, disabled by default. When calling editor.setValue() and each field that was given a value should be enabled.

### Actual behavior
When calling setValue() on the editor, all fields remain disabled even when a value has been added.

### Steps to reproduce the behavior
 - Create editor with show_opt_in enabled with a given schema.
 - Call editor.setValue() to set values of the given schema.

If this is intended behavior, I'm not sure why one would want to set values, but then not have them show up in the json. If the goal is to have a default value in the opted-out fields, then a default should be set in a schema rather than using setValue().

**Comment by EasyPP** [comment]
I agree.

This bug, actually renders the opt_in feature useless for existing data, while it is still good while creating new data.

**Comment by EasyPP** [comment]
I would expect a fix could be made in this line: https://github.com/json-editor/json-editor/blob/9dc8413cd3101d0a1dee65004fa2195bb4496430/src/editors/object.js#L1211

maybe to something like this ?  :

```
    Object.entries(value).forEach(([i, val]) => {
      if (!this.cached_editors[i]) {
        this.addObjectProperty(i)
        if (this.editors[i]) {
             this.editors[i].setValue(val, initial, !!this.editors[i].template);
             this.editors[i].activate();
        }
      }
    })
```

**Comment by germanbisurgi** [comment]
fixed in #943
